features/netsocket/InternetSocket.cpp

```diff
@@ -174,6 +174,9 @@ nsapi_error_t InternetSocket::setsockopt(int level, int optname, const void *opt
         ret = NSAPI_ERROR_NO_SOCKET;
     } else {
         ret = _stack->setsockopt(_socket, level, optname, optval, optlen);
+        if (optname == NSAPI_BIND_TO_DEVICE && level == NSAPI_SOCKET) {
+            strncpy(_interface_name, static_cast<const char *>(optval), optlen);
+        }
```

Comment:
We should get cloud client input on this as well. We make the interface binding (of client socket) with options, where as linux uses the bind operation. Now they need to adapt to both options

Reply:
@JanneKiiskila Mind having someone chime in?

Reply:
@yogpan01   @TeroJaasko  @teetak01  @anttiylitokola  to check it up, please.

Reply:
This is now ready for integration, is there anything blocking ?

Reply:
If it helps in any way, the [mbed-client-testapp CI is passing fine](https://jenkins-internal.mbed.com/blue/organizations/jenkins/ARMmbed%2Fmbed-client-testapp/detail/PR-1238/9/pipeline/) with the most recent commit in this PR